public/canvas_tools/tools.js

```diff
@@ -375,18 +380,21 @@ function canvasClick(event) {
       addToUndo(undoGraphStates, graph);
       let newNode = new NodeData(0, x, y);
       graph.addNode(newNode);
+      GraphController.addNode(0, x, y);
       basicTool.state.stillInNode = true;
-      refreshHtml(graph, toolState);
+      refreshHtml(graph.nodeCount, graph.edgeCount, toolState, calculateGraphType(graph), graph.getAdjList());
     } else if (!basicTool.state.edgeMode) {
       enterBasicEdgeMode(nodeClicked);
     } else if (nodeClicked && nodeClicked != basicTool.state.edgeStart) {
       // add edge
       if (!graph.containsEdge(basicTool.state.edgeStart, nodeClicked)) {
         addToUndo(undoGraphStates, graph);
         graph.addEdge(basicTool.state.edgeStart, nodeClicked);
```

Comment:
This addEdge does not currently work because `counter` changes with time. The reason it works with the JS graph class is it's storing a reference to the object, not a case class like in Scala. So we need to rethink things but will do in a separate feature branch since the point of this one is to proof of concept using the Scala library in the JS code.